src/InstrumentationEngine/InstructionGraph.cpp

```diff
@@ -17,6 +17,15 @@ MicrosoftInstrumentationEngine::CInstructionGraph::~CInstructionGraph()
         CCriticalSectionHolder lock(&m_cs);
 
         CComPtr<CInstruction> pInstr = (CInstruction*)m_pFirstInstruction;
+
+        m_pFirstInstruction.Release();
+        m_pOrigFirstInstruction.Release();
+        
+        if(m_pUninstrumentedFirstInstruction != nullptr)
+        {
+            m_pUninstrumentedFirstInstruction.Release();
+        }
```

Comment:
You could also just do the following:

m_pFirstInstruction = nullptr;
m_OirigFirstInstruction = nullptr;
m_pUninstrumentedFirst = nullptr;

That way, you don't need to check m_pUninstrumentedFirstInstruction to see if it is null.